sensor/common/compliance/node_inventory_handler_test.go

```diff
@@ -136,6 +139,59 @@ func (s *NodeInventoryHandlerTestSuite) TestHandlerStopIgnoresError() {
 	s.NoError(h.Stopped().Wait())
 }
 
+// This test simulates a running Sensor loosing connection to Central, followed by a reconnect.
+// As soon as Sensor enters offline mode, it should send NACKs to Compliance.
+// In online mode, inventories are forwarded to Central, which responds with an ACK, that is passed to Compliance.
+func (s *NodeInventoryHandlerTestSuite) TestHandlerOfflineACKNACK() {
+	ch, producer := s.generateTestInputNoClose(1)
+	defer close(ch)
+	h := NewNodeInventoryHandler(ch, &mockAlwaysHitNodeIDMatcher{})
+	h.Notify(common.SensorComponentEventCentralReachable)
+	s.NoError(h.Start())
+	mockCentralAck(h, s)
+	stats1 := consumeAndCountCompliance(h.ComplianceC(), 1)
+	s.NoError(stats1.sc.Stopped().Wait())
+	s.Equal(1, stats1.ACKCount)
+
+	h.Notify(common.SensorComponentEventOfflineMode)
+	ch <- fakeNodeInventory(fmt.Sprintf("Node-%d", 4))
+	stats2 := consumeAndCountCompliance(h.ComplianceC(), 1)
+	s.NoError(stats2.sc.Stopped().Wait())
+	s.Equal(1, stats2.NACKCount)
+
+	h.Notify(common.SensorComponentEventCentralReachable)
+	ch <- fakeNodeInventory(fmt.Sprintf("Node-%d", 4))
+	mockCentralAck(h, s)
+	stats3 := consumeAndCountCompliance(h.ComplianceC(), 1)
+	s.NoError(stats3.sc.Stopped().Wait())
+	s.Equal(1, stats3.ACKCount)
+
+	s.NoError(producer.Stopped().Wait())
+
+	h.Stop(nil)
+	s.T().Logf("waiting for handler to stop")
+	s.NoError(h.Stopped().Wait())
+}
+
+func mockCentralAck(h *nodeInventoryHandlerImpl, s *NodeInventoryHandlerTestSuite) {
```

Comment:
I think, it would be better for this function to return an error instead taking `s *NodeInventoryHandlerTestSuite` as a parameter. It would make it more independent.

```suggestion
func mockCentralAck(h *nodeInventoryHandlerImpl) error {
```

Reply:
+1